src/actions_eval.ts

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import * as _vscode from "vscode";
+
+import { createLogger } from "./logger";
+
+// @ts-ignore
+const logger = createLogger("eval");
+
+// for some reason, the name used in the import statement is not always visible to the code being run with eval()
+// but global variables/constants are always visible.
+// @ts-ignore
+const vscode = _vscode;
+
+/**
+ * Execute javascript code passed from lua in an async function context
+ *
+ * - the variable `vscode` can be used to access the VSCode API
+ * - the variable `args` can be used to access the arguments passed from lua
+ *
+ * @param code the code to evaluate
+ * @param args arguments passed from lua
+ *
+ * @returns the result of evaluating the code, serialized to send back to lua
+ */
+export async function eval_for_client(code: string, args: any): Promise<any> {
```

Comment:
`args` is not used, is it still needed?

Reply:
It is 'used' in the sense that it is accessible from the code being evaluated. This is used to pass data from lua. There are examples of this in the unit tests.